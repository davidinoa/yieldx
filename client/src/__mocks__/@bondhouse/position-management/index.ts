import {
  CreateFundTransferResponse,
  CreateOrReplaceExternalPortfolioResponse,
  LedgerEntryStatusEnum,
  PortfolioStatus,
  AppType
} from "@bondhouse/position-management";

import samplePortfolio from "util/samplePortfolio";
import usdCash from "util/usdCash";

export function PortfoliosApi() {
  return {
    async internalPortfolioFundTransfer() {
      return {
        accepted: true,
        ledgerEntry: {
          id: "",
          tradeId: "",
          portfolioId: "",
          assetId: "",
          quantity: 1,
          tradePrice: 1,
          counterPartyId: "",
          relatedPositions: [],
          status: LedgerEntryStatusEnum.SETTLED
        }
      } as CreateFundTransferResponse;
    },

    updatePortfolio: jest.fn(async () => ({
      id: "mockId",
      createdAt: new Date().toISOString(),
      updateAt: new Date().toISOString(),
      status: PortfolioStatus.TERMINATED,
      currency: "USD",
      name: "Mock Portfolio",
      positions: [usdCash]
    })),

    getPortfolio: jest.fn(async () => {
      return {
        data: {
          portfolio: {
            ...samplePortfolio,
            positions: []
          }
        }
      };
    }),

    createPortfolioForInvestor: jest.fn(async () => {
      const portfolio = {
        ...samplePortfolio,
        positions: []
      };
      delete portfolio.metadata;
      return { data: { portfolio } };
    })
  };
}

export function TradesApi() {
  return {
    getTrades: jest.fn(async () => ({
      data: {
        ledgerEntries: []
      }
    }))
  };
}

export function ExternalPortfolioApi() {
  return {
    createExternalPortfolio: jest.fn(async () => {
      return {
        created: true,
        updated: false
      } as CreateOrReplaceExternalPortfolioResponse;
    }),
    getExternalPortfolio: jest.fn(async () => {
      return {
        data: {
          portfolio: { positions: [] }
        }
      };
    })
  };
}

export { AppType, PortfolioStatus, LedgerEntryStatusEnum };
