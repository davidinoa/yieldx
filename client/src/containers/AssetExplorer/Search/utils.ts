interface InGridFilter {
  filter: string | number;
  filterTo: number;
  filterType: string;
  type: string;
  dateFrom: string;
  dateTo: string;
  values: string[];
}

export interface AgGridFilters {
  [key: string]: any;
}

const getFilter = (filter: InGridFilter) => {
  /**
   *  The filter object from Ag Grid API. In the case of filtering `name` column.
   *  { filter: "Asset Name", filterType: "text", type: "contains"}
   */
  let searchFilter = {};
  const {
    filter: filterContent,
    filterType,
    type,
    filterTo,
    values,
    dateFrom,
    dateTo
  } = filter;
  if (filterType === "text") {
    switch (type) {
      case "contains":
        searchFilter = { containsText: filterContent };
        break;
      case "notContains":
        searchFilter = { doesNotContainText: filterContent };
        break;
      case "equals":
        searchFilter = { equalsText: filterContent };
        break;
      case "notEqual":
        searchFilter = { doesNotEqualText: filterContent };
        break;
      case "startsWith":
        searchFilter = { startsWithText: filterContent };
        break;
      case "endsWith":
        searchFilter = { endsWith: filterContent };
        break;
      default:
        searchFilter = { containsText: filterContent };
        break;
    }
  } else if (filterType === "number") {
    switch (type) {
      case "equals":
        searchFilter = { equalsNumber: filterContent };
        break;
      case "notEqual":
        searchFilter = { doesNotEqualNumber: filterContent };
        break;
      case "lessThan":
        searchFilter = { isLessThanNumber: filterContent };
        break;
      case "lessThanOrEqual":
        searchFilter = { isLessThanOrEqualToNumber: filterContent };
        break;
      case "greaterThan":
        searchFilter = { isGreaterThanNumber: filterContent };
        break;
      case "greaterThanOrEqual":
        searchFilter = { isGreaterThanOrEqualToNumber: filterContent };
        break;
      case "inRange":
        searchFilter = {
          isGreaterThanOrEqualToNumber: filterContent,
          isLessThanOrEqualToNumber: filterTo
        };
        break;
      default:
        searchFilter = { equalsNumber: filterContent };
        break;
    }
  } else if (filterType === "date") {
    const convertedDateFrom = new Date(dateFrom);
    const convertedDateTo = new Date(dateTo);
    switch (type) {
      case "equals":
        searchFilter = { equalsDate: convertedDateFrom };
        break;
      case "greaterThan":
        searchFilter = { isGreaterThanDate: convertedDateFrom };
        break;
      case "lessThan":
        searchFilter = { isLessThanDate: convertedDateFrom };
        break;
      case "notEqual":
        searchFilter = { doesNotEqualDate: convertedDateFrom };
        break;
      case "inRange":
        searchFilter = {
          isGreaterThanOrEqualToDate: convertedDateFrom,
          isLessThanOrEqualToDate: convertedDateTo
        };
        break;
      default:
        searchFilter = { equalsDate: convertedDateFrom };
        break;
    }
  } else if (filterType === "set" && values) {
    // We may need to add `set` type Ag grid column filter params like cusip or isin
    // in the search request in the future.
    // searchFilter = { inText: values };
  }
  return searchFilter;
};

export const convertInGridFilters = (filters: AgGridFilters) => {
  const convertedFilters: { [index: string]: any } = {};
  Object.keys(filters).forEach(key => {
    const inGridFilter = filters[key];
    if (key === "minimumQuantity") key = "bond.minimumDenomination";
    convertedFilters[key] = getFilter(inGridFilter);
  });

  return convertedFilters;
};
