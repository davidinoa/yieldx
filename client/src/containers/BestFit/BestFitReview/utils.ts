import { getRatingValue } from "../../../util/ratings";
import YieldXPortfolio, {
  YieldXPosition
} from "../../../models/YieldXPortfolio";

export function getBondRatingsData(
  portfolio: YieldXPortfolio
): { name: string; value: number }[] {
  if (portfolio.getBondSleeve) {
    const ratingMarketValues: { [key: string]: number } = {};
    const bondPortfolio = portfolio.getBondSleeve();
    bondPortfolio.positions.forEach((position: YieldXPosition) => {
      const { asset, weight } = position;
      const rating = asset.rating ?? "NR";
      ratingMarketValues[rating] = (ratingMarketValues[rating] || 0) + weight;
    });
    return Object.entries(ratingMarketValues)
      .map(([rating, weight]) => {
        return {
          name: rating,
          value: weight
        };
      })
      .sort((a, b) => {
        return getRatingValue(a.name) - getRatingValue(b.name);
      });
  }
  return [];
}
