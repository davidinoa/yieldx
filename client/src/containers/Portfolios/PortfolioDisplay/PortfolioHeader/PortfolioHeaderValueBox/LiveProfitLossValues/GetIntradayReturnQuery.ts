import { gql } from "@apollo/client";

export default gql`
  query calculatePortfolioIntradayReturn(
    $calculateIntradayReturnRequestInput: CalculateIntradayReturnRequestInput
  ) {
    calculatePortfolioIntradayReturn(
      calculateIntradayReturnRequestInput: $calculateIntradayReturnRequestInput
    ) {
      marketValueReturn
      percentReturn
      totalReturn
    }
  }
`;
