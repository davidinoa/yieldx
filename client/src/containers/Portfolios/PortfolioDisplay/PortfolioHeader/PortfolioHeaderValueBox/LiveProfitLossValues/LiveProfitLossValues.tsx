import React from "react";
import { Row } from "react-bootstrap";
import {
  PortfolioInput,
  useRoverPortfolioAnalyzerCalculateIntradayReturnQuery
} from "providers/graphql/hooks";
import { ValuePnL } from "../../PortfolioHeader.styles";
import ProfitLossValue from "./ProfitLossValue/ProfitLossValue";

interface Props {
  portfolioInput: PortfolioInput;
}

function LiveProfitLossValues({ portfolioInput }: Props) {
  const { data, error } = useRoverPortfolioAnalyzerCalculateIntradayReturnQuery(
    {
      variables: {
        calculateIntradayReturnRequestInput: {
          portfolio: portfolioInput
        }
      }
    }
  );

  const gainToday =
    data?.roverPortfolioAnalyzerCalculateIntradayReturn?.marketValueReturn;
  return (
    <Row>
      <ValuePnL>
        {/*
          <ProfitLossValue value={gainTotal} label="total gain/loss:" />
        */}
        <ProfitLossValue
          value={gainToday}
          label="today gain/loss:"
          error={Boolean(error)}
        />
      </ValuePnL>
    </Row>
  );
}

export default LiveProfitLossValues;
