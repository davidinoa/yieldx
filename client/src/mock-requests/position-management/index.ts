import { createGetPortfolioHandler } from "./createGetPortfolioHandler";

export default {
  defaultGetPortfolio: createGetPortfolioHandler
};
