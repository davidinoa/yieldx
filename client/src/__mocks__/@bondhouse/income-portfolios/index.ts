import {
  AppType,
  NewProposalRequest,
  OrderDirectionEnum,
  PortfolioStatus,
  PortfolioType,
  GetComparisonsResponse
} from "@bondhouse/income-portfolios";
import { MockResponse } from "test/utils";

// eslint-disable-next-line @typescript-eslint/no-empty-function,no-shadow
function ProposalsApi() {}

ProposalsApi.prototype.getComparisons = jest.fn(() =>
  Promise.resolve<MockResponse<GetComparisonsResponse>>({
    data: {
      comparisons: [
        {
          comparisonType: "Risk",
          inpaas: 16.812674797951423,
          average: 37.5,
          difference: 20.687325202048577,
          percentageDifference: 55.166200538796204
        },
        {
          comparisonType: "Expense Ratio",
          inpaas: 1.5599931630856463,
          average: 2.99,
          difference: 1.430006836914354,
          percentageDifference: 47.82631561586468
        },
        {
          comparisonType: "Max Drawdown",
          inpaas: -19.50881468122584,
          average: -32.6,
          difference: 13.091185318774162,
          percentageDifference: 40.157010180288836
        }
      ]
    }
  })
);
ProposalsApi.prototype.newProposal = jest.fn(
  (newProposalRequest?: NewProposalRequest) =>
    Promise.resolve({
      data: {
        proposal: {
          proposalId: "d646831d-2e18-43a9-9a98-9c12e97cd2f7",
          status: "PENDING",
          updatedAt: "2020-05-22T12:48:51.233506Z",
          commentary: [
            {
              topic: "Income",
              content:
                "The expected monthly income is $1008 for an annualized yield of 6.05%."
            },
            {
              topic: "Risk",
              content:
                "You have a 29% chance of losing money in a given year, and the portfolio's largest drop in account value was 14.2% in the last five years. This is -4.4% less risk than similar securities yielding 6-7%."
            },
            {
              topic: "Expenses",
              content:
                "The weighted expense ratio is 0.41% per year. This is 1.12% less than similar funds with this yield."
            }
          ],
          portfolio: {
            id: "e4d4c1be-ad5b-4221-8e34-12e0e8d73dcf",
            currency: "USD",
            name: "Your new portfolio",
            positions: [
              {
                _yield: 0.77,
                id: "a7524667._04b96c46.buy",
                assetId: "fund",
                publicIdentifiers: {
                  cusip: "46434V878",
                  isin: "US46434V8789",
                  ticker: "ICSH"
                },
                quantity: 5,
                marketValue: 252.60000000000002,
                weight: 25.260000000000005,
                price: 50.52,
                name: "iShares Ultra Short-Term Bond ETF"
              },
              {
                _yield: 0,
                id: "USD",
                assetId: "USD",
                publicIdentifiers: {},
                quantity: 163.80999999999995,
                marketValue: 163.80999999999995,
                weight: 16.380999999999997,
                price: 1,
                name: "Cash"
              }
            ],
            createdAt: "2020-05-22T12:48:51.233586Z",
            updatedAt: "2020-05-22T12:48:51.233586Z",
            metadata: { app: "inpaas" }
          },
          incomePortfoliosAnalysis: {
            income: 1007.5426586045,
            risk: 9.785923627675485,
            expenseRatio: 0.41044404884999997,
            yield: 6.045255951627,
            afterTaxYield: 3.6271535709762004,
            maxDrawdown: -14.161982640325931
          },
          proposalRequest: {
            goal: {
              income: newProposalRequest?.goal.income,
              // eslint-disable-next-line @typescript-eslint/ban-ts-comment
              // @ts-ignore
              yield: newProposalRequest?.goal.yield
            },
            investmentAmount: newProposalRequest?.investmentAmount
          },
          orders: [
            {
              assetId: "102dd159-1401-49dc-bad0-9cb776fbe5cb",
              direction: "BUY",
              quantity: 1209
            },
            {
              assetId: "6f5c3533-89be-48ab-806a-ab693cf0a7ff",
              direction: "BUY",
              quantity: 933.9999999999999
            },
            {
              assetId: "8feba4a4-1a13-47be-b447-1752287b8e2f",
              direction: "BUY",
              quantity: 398
            },
            {
              assetId: "4357f998-6755-40e6-8607-adbd454f9756",
              direction: "BUY",
              quantity: 217
            },
            {
              assetId: "a0135762-04b7-4dc6-8077-3ef61bcedeaa",
              direction: "BUY",
              quantity: 114
            },
            {
              assetId: "f10c6a1d-c607-4666-8621-8b1a2c3741b3",
              direction: "BUY",
              quantity: 489
            },
            {
              assetId: "e4871590-916e-4256-b957-18bb2e8e8b2f",
              direction: "BUY",
              quantity: 295.0000096246391
            },
            {
              assetId: "621e7f48-e332-4e92-a875-260ed4ec9569",
              direction: "BUY",
              quantity: 2787
            },
            {
              assetId: "ac1ab82c-c1c4-4d34-83fa-df78c79404bf",
              direction: "BUY",
              quantity: 714
            },
            {
              assetId: "USD",
              direction: "SELL",
              quantity: -199999.99999999997
            }
          ]
        }
      }
    })
);

ProposalsApi.prototype.getProposal = jest.fn((proposalId: string) =>
  Promise.resolve({
    data: {
      proposal: {
        proposalId,
        status: "PENDING",
        updatedAt: "2020-05-22T12:48:51.233506Z",
        commentary: [
          {
            topic: "Income",
            content:
              "The expected monthly income is $1008 for an annualized yield of 6.05%."
          },
          {
            topic: "Risk",
            content:
              "You have a 29% chance of losing money in a given year, and the portfolio's largest drop in account value was 14.2% in the last five years. This is -4.4% less risk than similar securities yielding 6-7%."
          },
          {
            topic: "Expenses",
            content:
              "The weighted expense ratio is 0.41% per year. This is 1.12% less than similar funds with this yield."
          }
        ],
        portfolio: {
          id: "e4d4c1be-ad5b-4221-8e34-12e0e8d73dcf",
          currency: "USD",
          name: "Your new portfolio",
          positions: [
            {
              _yield: 0.77,
              id: "a7524667._04b96c46.buy",
              assetId: "fund",
              publicIdentifiers: {
                cusip: "46434V878",
                isin: "US46434V8789",
                ticker: "ICSH"
              },
              quantity: 5,
              marketValue: 252.60000000000002,
              weight: 25.260000000000005,
              price: 50.52,
              name: "iShares Ultra Short-Term Bond ETF"
            },
            {
              _yield: 0,
              id: "USD",
              assetId: "USD",
              publicIdentifiers: {},
              quantity: 163.80999999999995,
              marketValue: 163.80999999999995,
              weight: 16.380999999999997,
              price: 1,
              name: "Cash"
            }
          ],
          createdAt: "2020-05-22T12:48:51.233586Z",
          updatedAt: "2020-05-22T12:48:51.233586Z",
          metadata: { app: "inpaas" }
        },
        incomePortfoliosAnalysis: {
          income: 1007.5426586045,
          risk: 9.785923627675485,
          expenseRatio: 0.41044404884999997,
          yield: 6.045255951627,
          afterTaxYield: 3.6271535709762004,
          maxDrawdown: -14.161982640325931
        },
        proposalRequest: {
          goal: {
            income: 100_000,
            // eslint-disable-next-line @typescript-eslint/ban-ts-comment
            // @ts-ignore
            yield: 10
          },
          investmentAmount: 1_000_000
        },
        orders: [
          {
            assetId: "102dd159-1401-49dc-bad0-9cb776fbe5cb",
            direction: "BUY",
            quantity: 1209
          },
          {
            assetId: "6f5c3533-89be-48ab-806a-ab693cf0a7ff",
            direction: "BUY",
            quantity: 933.9999999999999
          },
          {
            assetId: "8feba4a4-1a13-47be-b447-1752287b8e2f",
            direction: "BUY",
            quantity: 398
          },
          {
            assetId: "4357f998-6755-40e6-8607-adbd454f9756",
            direction: "BUY",
            quantity: 217
          },
          {
            assetId: "a0135762-04b7-4dc6-8077-3ef61bcedeaa",
            direction: "BUY",
            quantity: 114
          },
          {
            assetId: "f10c6a1d-c607-4666-8621-8b1a2c3741b3",
            direction: "BUY",
            quantity: 489
          },
          {
            assetId: "e4871590-916e-4256-b957-18bb2e8e8b2f",
            direction: "BUY",
            quantity: 295.0000096246391
          },
          {
            assetId: "621e7f48-e332-4e92-a875-260ed4ec9569",
            direction: "BUY",
            quantity: 2787
          },
          {
            assetId: "ac1ab82c-c1c4-4d34-83fa-df78c79404bf",
            direction: "BUY",
            quantity: 714
          },
          {
            assetId: "USD",
            direction: "SELL",
            quantity: -199999.99999999997
          }
        ]
      }
    }
  })
);

ProposalsApi.prototype.getMinMaxYield = jest.fn(() =>
  Promise.resolve({
    data: {
      maxYield: 14,
      minYield: 1
    }
  })
);

ProposalsApi.prototype.getMinMaxRisk = jest.fn(() =>
  Promise.resolve({
    data: {
      maxRisk: 12,
      minRisk: 1
    }
  })
);

ProposalsApi.prototype.rebalanceProposal = jest.fn(() =>
  Promise.resolve({
    data: {
      proposal: {
        proposalId: "d646831d-2e18-43a9-9a98-9c12e97cd2f7",
        status: "PENDING",
        updatedAt: "2020-05-22T12:48:51.233506Z",
        commentary: [
          {
            topic: "Income",
            content:
              "The expected monthly income is $1008 for an annualized yield of 6.05%."
          },
          {
            topic: "Risk",
            content:
              "You have a 29% chance of losing money in a given year, and the portfolio's largest drop in account value was 14.2% in the last five years. This is -4.4% less risk than similar securities yielding 6-7%."
          },
          {
            topic: "Expenses",
            content:
              "The weighted expense ratio is 0.41% per year. This is 1.12% less than similar funds with this yield."
          }
        ],
        portfolio: {
          id: "e4d4c1be-ad5b-4221-8e34-12e0e8d73dcf",
          currency: "USD",
          name: "Your new portfolio",
          positions: [
            {
              _yield: 0.77,
              id: "a7524667._04b96c46.buy",
              assetId: "fund",
              publicIdentifiers: {
                cusip: "46434V878",
                isin: "US46434V8789",
                ticker: "ICSH"
              },
              quantity: 5,
              marketValue: 252.60000000000002,
              weight: 25.260000000000005,
              price: 50.52,
              name: "iShares Ultra Short-Term Bond ETF"
            },
            {
              _yield: 0,
              id: "USD",
              assetId: "USD",
              publicIdentifiers: {},
              quantity: 163.80999999999995,
              marketValue: 163.80999999999995,
              weight: 16.380999999999997,
              price: 1,
              name: "Cash"
            }
          ],
          createdAt: "2020-05-22T12:48:51.233586Z",
          updatedAt: "2020-05-22T12:48:51.233586Z",
          metadata: { app: "inpaas" }
        },
        incomePortfoliosAnalysis: {
          income: 1007.5426586045,
          risk: 9.785923627675485,
          expenseRatio: 0.41044404884999997,
          yield: 6.045255951627,
          afterTaxYield: 3.6271535709762004,
          maxDrawdown: -14.161982640325931
        },
        proposalRequest: {
          goal: {
            income: 1_000,
            // eslint-disable-next-line @typescript-eslint/ban-ts-comment
            // @ts-ignore
            yield: 10
          },
          investmentAmount: 100000
        },
        orders: [
          {
            assetId: "102dd159-1401-49dc-bad0-9cb776fbe5cb",
            direction: "BUY",
            quantity: 1209
          },
          {
            assetId: "6f5c3533-89be-48ab-806a-ab693cf0a7ff",
            direction: "BUY",
            quantity: 933.9999999999999
          },
          {
            assetId: "8feba4a4-1a13-47be-b447-1752287b8e2f",
            direction: "BUY",
            quantity: 398
          },
          {
            assetId: "4357f998-6755-40e6-8607-adbd454f9756",
            direction: "BUY",
            quantity: 217
          },
          {
            assetId: "a0135762-04b7-4dc6-8077-3ef61bcedeaa",
            direction: "BUY",
            quantity: 114
          },
          {
            assetId: "f10c6a1d-c607-4666-8621-8b1a2c3741b3",
            direction: "BUY",
            quantity: 489
          },
          {
            assetId: "e4871590-916e-4256-b957-18bb2e8e8b2f",
            direction: "BUY",
            quantity: 295.0000096246391
          },
          {
            assetId: "621e7f48-e332-4e92-a875-260ed4ec9569",
            direction: "BUY",
            quantity: 2787
          },
          {
            assetId: "ac1ab82c-c1c4-4d34-83fa-df78c79404bf",
            direction: "BUY",
            quantity: 714
          },
          {
            assetId: "USD",
            direction: "SELL",
            quantity: -199999.99999999997
          }
        ]
      }
    }
  })
);

// eslint-disable-next-line @typescript-eslint/no-unused-vars,@typescript-eslint/no-empty-function
function PortfolioMappingsApi() {}

PortfolioMappingsApi.prototype.createOrReplacePortfolioMapping = jest.fn();
PortfolioMappingsApi.prototype.getPortfolioMapping = jest.fn(() =>
  Promise.resolve({
    data: {
      portfolioMapping: {
        institution: "bf8a5597-9569-4510-babd-a6d9655e9f0d",
        proposalId: "proposalId",
        portfolioId: "portfolioId"
      }
    }
  })
);

export {
  ProposalsApi,
  PortfolioMappingsApi,
  OrderDirectionEnum,
  PortfolioType,
  PortfolioStatus,
  AppType
};
