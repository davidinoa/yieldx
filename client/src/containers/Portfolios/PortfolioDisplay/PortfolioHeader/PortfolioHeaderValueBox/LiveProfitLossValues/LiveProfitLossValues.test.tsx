import React from "react";
import { buildMockPortfolio, render, screen, waitFor } from "test/utils";
import LiveProfitLossValues from "./LiveProfitLossValues";
import { setupServer } from "msw/node";
import { AppType } from "@bondhouse/position-management";
import { PortfolioType } from "@bondhouse/income-portfolios";
import toInputPortfolio from "util/toInputPortfolio";
import { waitForElementToBeRemoved } from "@testing-library/react";
import { client } from "providers/graphql";
import mockInpaasPortfolioRequests from "mock-requests/graphql/mockInpaasPortfolioRequests";
import createMockHandler from "mock-requests/graphql/createMockHandler";

const server = setupServer();
const portfolio = buildMockPortfolio({
  appType: AppType.INPAAS,
  portfolioType: PortfolioType.EXTERNAL
});
describe("LiveProfitLossValues", () => {
  beforeAll(() => server.listen());
  afterEach(() => client.clearStore());
  afterAll(() => server.close());
  beforeEach(() =>
    server.use(
      createMockHandler({
        roverPortfolioAnalyzerCalculateIntradayReturn: {
          marketValueReturn: 2.5,
          totalReturn: 2.5,
          percentReturn: 1,
          __typename: "CalculateIntradayReturnResponse"
        }
      })
    )
  );
  test("Should render", () => {
    const { container } = render(
      <LiveProfitLossValues portfolioInput={toInputPortfolio(portfolio)} />
    );
    expect(container).toBeInTheDocument();
  });

  test("Should display the value returned from getTodayGain, + $5.50 in this case", async () => {
    const { asFragment } = render(
      <LiveProfitLossValues portfolioInput={toInputPortfolio(portfolio)} />
    );
    await waitForElementToBeRemoved(() => screen.queryByTestId("spinner"));
    expect(await screen.findByText("+ $2.50")).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
});
