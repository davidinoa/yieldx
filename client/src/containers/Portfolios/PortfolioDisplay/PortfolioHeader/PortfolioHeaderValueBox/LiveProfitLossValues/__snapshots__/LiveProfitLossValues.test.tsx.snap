// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LiveProfitLossValues Should display the value returned from getTodayGain, + $5.50 in this case 1`] = `
<DocumentFragment>
  .c0 {
  padding-top: 0.85rem;
  border-top: 1px solid var(--border-new);
}

.c0 span.label {
  display: inherit;
  font-weight: bold;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.2em;
  -moz-letter-spacing: 0.2em;
  -ms-letter-spacing: 0.2em;
  letter-spacing: 0.2em;
  font-weight: bold;
  font-family: var(--font-primary);
  color: var(--subtext-3);
  font-size: 0.6rem;
}

.c0 span.value {
  display: inherit;
  text-align: center;
  font-family: var(--font-secondary);
  font-weight: normal;
  font-size: 0.7rem;
}

.c0 span.value.positive {
  color: var(--active);
}

.c0 span.value.negative {
  color: var(--issue);
}

.c0 .divider {
  border-right: 1px solid var(--border-new);
}

@media (min-width:1600px) {
  .c0 {
    padding-top: 1.25rem;
  }
}

@media (min-width:1600px) {
  .c0 span.label {
    font-size: 0.75rem;
  }
}

@media (min-width:1600px) {
  .c0 span.value {
    font-size: 0.85rem;
  }
}

<div
    class="row"
  >
    <div
      class="c0 col-24 d-flex justify-content-around"
    >
      <div
        class="d-inline-block performance"
      >
        <span
          class="label"
        >
          today gain/loss:
        </span>
        <span
          class="value positive"
        >
          + $2.50
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
